src/services/BaseService.ts

```diff
@@ -111,25 +111,30 @@ export default abstract class BaseService {
 	private createTransporter(platformId: string): nodeMailer.Transporter | firebaseMessaging.Messaging {
 		const config = Utils.getPlatformConfig(platformId);
 		const transporter = this._createTransporter(config);
-		const platformPushTransporter = {
-			platformId,
-			transporter,
-		};
-		this.transporters.push(platformPushTransporter);
+		// do not store transporter on multiple config
+		if (config.hasMultipleAccounts !== true) {
+			const platformPushTransporter = {
+				platformId,
+				transporter,
+			};
+			this.transporters.push(platformPushTransporter);
+		}
 		return transporter;
 	}
 
 	private getTransporter(platformId: string): nodeMailer.Transporter | firebaseMessaging.Messaging {
-		const currentTransporter: PlatformMailTransporter | PlatformPushTransporter | undefined = this.transporters.find(
-			(transporter: PlatformMailTransporter | PlatformPushTransporter) => {
-				return transporter.platformId === platformId;
-			},
-		);
-
-		if (currentTransporter) {
-			return currentTransporter.transporter;
+		const config = Utils.getPlatformConfig(platformId);
+		if (config.hasMultipleAccounts !== true) {
+			const currentTransporter: PlatformMailTransporter | PlatformPushTransporter | undefined = this.transporters.find(
+				(transporter: PlatformMailTransporter | PlatformPushTransporter) => {
+					return transporter.platformId === platformId;
+				},
+			);
+
+			if (currentTransporter) {
+				return currentTransporter.transporter;
```

Comment:
why not using a transporter unless it fails and then use the next one instead of spooky randomization? specific issues should block a transporter (wrong credentials, non-temporary issues...) for a specific time to on next change the transporter is used which has not been failed (nen temporarily) as longest?

Reply:
The transporter now stores information about the last successfull/failed request which are used to deactivate a unavailable transporter for one hour. If no active transporters are available a random one is retried.